hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/PathMetadata.java

```diff
@@ -71,6 +79,8 @@ public PathMetadata(S3AFileStatus fileStatus, Tristate isEmptyDir, boolean
     this.fileStatus = fileStatus;
     this.isEmptyDirectory = isEmptyDir;
     this.isDeleted = isDeleted;
+    // modtime is derived from file status
+    setLastUpdated(fileStatus.getModificationTime());
```

Comment:
We will update this when used, so it seems safe that way. But when creating a PM instance from a ddb record, I feel like we would overwrite the last_updated with the mod_time, won't we?